src/config.rs

```diff
@@ -0,0 +1,2402 @@
+        /// normal use. It contains all the `"warning"` content.
+        #[serde(rename(deserialize = "info"))]
+        Info,
+        /// `"warning"` : V2Ray has encountered some problems, usually
+        /// external problems, which do not affect the normal operation
+        /// of V2Ray, but may affect the user experience. It contains
+        /// all the `"error"` content.
+        #[serde(rename(deserialize = "warning"))]
+        Warning,
+        /// `"error"` : V2Ray has encountered a problem that cannot run
+        /// normally and needs to be resolved immediately.
+        #[serde(rename(deserialize = "error"))]
+        Error,
+        /// `"none"` : Do not record anything.
+        #[serde(rename(deserialize = "none"))]
+        None,
+    }
+
+    /// `LogObject` The corresponding configuration file `log` entries.
+    ///
+    /// ```json
+    /// {
+    ///     "access": "文件地址",",
+    ///     "error": "文件地址",",
+    ///     "loglevel": "warning",
+    /// }
+    /// ```
+    #[derive(Deserialize, Debug)]
+    #[serde(rename_all = "camelCase")]
+    pub struct Log {
+        /// The file address of the access log. Its value is a legal file
+        /// address, such as `"/var/log/v2ray/access.log"` (Linux) or
+        /// `"C:\\Temp\\v2ray\\_access.log"` (Windows). When this item is
+        /// not specified or is empty, it means that the log is output to
+        /// stdout. V2Ray 4.20 added a special value `none`, that is, close
+        /// the access log.
+        access: String,
+        /// The file address of the error log. Its value is a legal file
+        /// address, such as `"/var/log/v2ray/error.log"` (Linux) or
+        /// `"C:\\Temp\\v2ray\\_error.log"` (Windows). When this item
+        /// is not specified or is empty, it means that the log is output
+        /// to stdout. V2Ray 4.20 added a special value none, that is, close
+        /// the error log ( `loglevel: "none"` equal to).
+        error: String,
+        /// The level of the log. The default value is `"warning"`.
+        log_level: LogLevel,
+    }
+
+    #[cfg(test)]
+    mod tests {
+        use super::*;
+        use serde_json::{from_str, from_value, json};
+
+        #[test]
+        pub fn test_deserialize_level() {
+            let ds = from_str::<LogLevel>("\"warning\"");
+            assert_eq!(ds.is_ok(), true);
+            assert!(matches!(ds.unwrap(), LogLevel::Warning));
+
+            let ds = from_str::<LogLevel>("\"info\"");
+            assert!(matches!(ds.unwrap(), LogLevel::Info));
+        }
+
+        #[test]
+        pub fn test_deserialize_log() {
+            let result = from_value::<Log>(json!({
+                "access": "/var/log/v2ray/access.log",
+                "error": "/var/log/v2ray/error.log",
+                "logLevel": "info"
+            }));
+
+            assert!(result.is_ok());
+            assert_eq!(result.unwrap().access, "/var/log/v2ray/access.log");
+        }
+    }
+}
+
+pub mod api {
```

Comment:
Consider separating these mods into different files. 2402 lines seems too much for a single file.